src/spandrel/__helpers/model_descriptor.py

```diff
@@ -124,17 +124,52 @@ def __init__(
 
         self.model.load_state_dict(state_dict)  # type: ignore
 
-    def to(self, device: torch.device):
-        self.model.to(device)
+    def to(
+        self,
+        device: int | torch.device | None = torch.device("cpu"),
+        dtype: torch.dtype | str | None = None,
+        non_blocking: bool = False,
+    ):
+        self.model.to(device, dtype, non_blocking)
+        return self
+
+    def half(self):
+        self.model.half()
+        return self
+
+    def float(self):
+        self.model.float()
         return self
 
+    def cuda(self):
+        self.model.cuda()
+        return self
+
+    def cpu(self):
+        self.model.cpu()
+        return self
+
+    def eval(self):
+        self.model.eval()
+        return self
+
+    def __call__(self, *args, **kwargs):  # noqa: ANN002
+        return self.model(*args, **kwargs)
```

Comment:
No. This throws away any semblance of type safety.

Reply:
I'll make it throw an error then

Reply:
Why? Just remove it.